lib/Chat/Notifier.php

```diff
@@ -426,27 +428,27 @@ protected function getDefaultGroupNotification(): int {
 	 *
 	 * @param string $userId
 	 * @param IComment $comment
+	 * @param Room $room
+	 * @param Attendee|null $attendee
 	 * @return bool
 	 */
-	protected function shouldMentionedUserBeNotified(string $userId, IComment $comment): bool {
+	protected function shouldMentionedUserBeNotified(string $userId, IComment $comment, Room $room, ?Attendee $attendee = null): bool {
 		if ($comment->getActorType() === Attendee::ACTOR_USERS && $userId === $comment->getActorId()) {
 			// Do not notify the user if they mentioned themselves
 			return false;
 		}
 
-		if (!$this->userManager->userExists($userId)) {
-			return false;
-		}
```

Comment:
[`getActorsByType()`](https://github.com/nextcloud/spreed/blob/351c3090c073a7208c830447ac7807bf916d2718/lib/Model/AttendeeMapper.php#L105) does not ensure that the user actually exists. Therefore, shouldn't this check be kept for all users instead of doing it only for those without an already loaded attendee?

Reply:
Yeah, that is exactly the point. The user deletion removes all user attendees already so anything like that is unexpected and at the same time doesn't cause any harm so we can save the O(n) database queries.